### PR TITLE
Make max_stream_bidi configurable

### PR DESCRIPTION
**Pull request by deweerdt**

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -1654,10 +1654,10 @@ static int on_config_listen(h2o_configurator_command_t *cmd, h2o_configurator_co
                 listener = add_listener(fd, ai->ai_addr, ai->ai_addrlen, ctx->hostconf == NULL, 0);
                 listener->quic.ctx = quic;
                 if (quic_node != NULL) {
-                    yoml_t **retry_node, **sndbuf, **rcvbuf, **amp_limit, **qpack_encoder_table_capacity;
+                    yoml_t **retry_node, **sndbuf, **rcvbuf, **amp_limit, **qpack_encoder_table_capacity, **max_stream_data_bidi;
```
Comment (kazuho) [inline src/main.c:1657]: I think you meant `max_streams_bidi`?

"max_stream_data" and "max_streams" are different things in QUIC (see section [19.10](https://quicwg.org/base-drafts/draft-ietf-quic-transport.html#name-max_stream_data-frames) and 19.11 of QUIC transport draft).

File: src/main.c [hunk 2]
```diff
@@ -1654,10 +1654,10 @@ static int on_config_listen(h2o_configurator_command_t *cmd, h2o_configurator_co
...
                     if (h2o_configurator_parse_mapping(
-                            cmd, *quic_node, NULL, "retry:s,sndbuf:s,rcvbuf:s,amp-limit:s,qpack-encoder-table-capacity:s",
-                            &retry_node, &sndbuf, &rcvbuf, &amp_limit, &qpack_encoder_table_capacity) != 0)
+                            cmd, *quic_node, NULL, "retry:s,sndbuf:s,rcvbuf:s,amp-limit:s,qpack-encoder-table-capacity:s,max-stream-data-bidi:s",
```
Comment (kazuho) [inline src/main.c:1659]: Here also.

File: src/main.c (hunk 1 above)
Comment (deweerdt) [reply src/main.c]: I did, thank you!

File: src/main.c (hunk 2 above)
Comment (deweerdt) [reply src/main.c]: Fixed in 8badc66

File: src/main.c [hunk 3]
```diff
@@ -1654,10 +1654,10 @@ static int on_config_listen(h2o_configurator_command_t *cmd, h2o_configurator_co
                 listener = add_listener(fd, ai->ai_addr, ai->ai_addrlen, ctx->hostconf == NULL, 0);
                 listener->quic.ctx = quic;
                 if (quic_node != NULL) {
-                    yoml_t **retry_node, **sndbuf, **rcvbuf, **amp_limit, **qpack_encoder_table_capacity, **max_stream_data_bidi;
+                    yoml_t **retry_node, **sndbuf, **rcvbuf, **amp_limit, **qpack_encoder_table_capacity, **max_stream_bidi;
```
Comment (kazuho) [inline src/main.c:1657]:
```suggestion
                    yoml_t **retry_node, **sndbuf, **rcvbuf, **amp_limit, **qpack_encoder_table_capacity, **max_streams_bidi;
```

File: src/main.c [hunk 4]
```diff
@@ -1654,10 +1654,10 @@ static int on_config_listen(h2o_configurator_command_t *cmd, h2o_configurator_co
...
                     if (h2o_configurator_parse_mapping(
-                            cmd, *quic_node, NULL, "retry:s,sndbuf:s,rcvbuf:s,amp-limit:s,qpack-encoder-table-capacity:s,max-stream-data-bidi:s",
-                            &retry_node, &sndbuf, &rcvbuf, &amp_limit, &qpack_encoder_table_capacity, &max_stream_data_bidi) != 0)
+                            cmd, *quic_node, NULL, "retry:s,sndbuf:s,rcvbuf:s,amp-limit:s,qpack-encoder-table-capacity:s,max-stream-bidi:s",
+                            &retry_node, &sndbuf, &rcvbuf, &amp_limit, &qpack_encoder_table_capacity, &max_stream_bidi) != 0)
```
Comment (kazuho) [inline src/main.c:1660]:
```suggestion
                            cmd, *quic_node, NULL, "retry:s,sndbuf:s,rcvbuf:s,amp-limit:s,qpack-encoder-table-capacity:s,max-streams-bidi:s",
                            &retry_node, &sndbuf, &rcvbuf, &amp_limit, &qpack_encoder_table_capacity, &max_streams_bidi) != 0)
```

File: src/main.c [hunk 5]
```diff
@@ -1689,8 +1689,8 @@ static int on_config_listen(h2o_configurator_command_t *cmd, h2o_configurator_co
                                                    &listener->quic.qpack.encoder_table_capacity) != 0)
                             return -1;
                     }
-                    if (max_stream_data_bidi != NULL) {
-                        if (h2o_configurator_scanf(cmd, *max_stream_data_bidi, "%" SCNu64,
+                    if (max_stream_bidi != NULL) {
+                        if (h2o_configurator_scanf(cmd, *max_stream_bidi, "%" SCNu64,
```
Comment (kazuho) [inline src/main.c:1693]:
```suggestion
                    if (max_streams_bidi != NULL) {
                        if (h2o_configurator_scanf(cmd, *max_streams_bidi, "%" SCNu64,
```

**Review by kazuho — COMMENTED** [review]
LGTM modulo the name below. I am not sure if QUIC WG chose the right name, but for whatever reason it is plural.

**Comment by deweerdt** [comment]
Thank you, amended

**Comment by kazuho** [comment]
Clicked one suggestion that was left open, that I think have held the build from succeeding. Let's see how it goes.

**Comment by deweerdt** [comment]
gah, thank you. The build did succeed

**Pull request merged**